sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2030,7 +2030,25 @@ class Dataset[T] private[sql](
    * @group typedrel
    * @since 2.3.0
    */
-  def unionByName(other: Dataset[T]): Dataset[T] = withSetOperator {
+  def unionByName(other: Dataset[T]): Dataset[T] = unionByName(other, false)
+
+  /**
+   * Returns a new Dataset containing union of rows in this Dataset and another Dataset.
+   *
+   * This is different from both `UNION ALL` and `UNION DISTINCT` in SQL. To do a SQL-style set
+   * union (that does deduplication of elements), use this function followed by a [[distinct]].
```

Comment:
this is not true now.

Reply:
Actually in original `unionByName`, its doc has this section too:

> This is different from both `UNION ALL` and `UNION DISTINCT` in SQL. To do a SQL-style set
> union (that does deduplication of elements), use this function followed by a [[distinct]].

Re-read this doc, even with original `unionByName` behavior, it is a bit confusing to me. Do you think we should remove "To do a SQL-style set union (that does deduplication of elements), use this function followed by a [[distinct]]."?

Reply:
Wait really? When did we change the semantics? What was confusing about that documentation? (it was added because users were confused by the behavior...)

Reply:
I read `To do a SQL-style set union`, it sounds like if you add `distinct`, you will get a SQL-style union. But it behaves different to SQL union at all.

Reply:
Seems like we mistakenly copied the doc from `union` to `unionByName`.